Tools/PRWatcher.ps1

```diff
@@ -0,0 +1,190 @@
+				[System.Version]$prVersion = $title[$prVerLoc]
+				#Write-Debug 0 $title[$prVerLoc]
+			}catch {
+				try {
+					[System.Version]$prVersion = $title[-2]
+				#Write-Debug 1 $title[-2]
+				}catch {
+					try {
+						[System.Version]$prVersion = $title[-3]
+				#Write-Debug 2 $title[-3]
+					}catch {
+						try {
+							[System.Version]$prVersion = $title[-4]
+				#Write-Debug 3 $title[-4]
+						}catch {
+							$prVersion = $title[-2]
+				#Write-Debug 4
+						}
+					}
+				}
+			}
+			#Store version number
+			if ($prVersion -eq "") {
+				$prVersion = $carryClip 
+			}
+			
+			$validColor = "green"
+			$invalidColor = "red"
+			$cautionColor = "yellow"
+			
+			#Get the PackageIdentifier out of the PR title, and alert if it matches the auth list.
+			$cleanOut = (Get-CleanClip); 
+			$AuthMatch = $AuthList.PackageIdentifier -match ($cleanOut.split("[.]")[0]+"."+$cleanOut.split("[.]")[1])
+			if ($AuthMatch) {
+				$AuthListLine = $AuthList | where {$_.PackageIdentifier -match $AuthMatch}
+				$strictness = $AuthListLine.strictness | sort -Unique
+				$AuthAccount = $AuthListLine.account | sort -Unique
+				$strictColor = ""
+				if ($strictness -eq "must") {
+					$strictColor = $invalidColor
+				} else {
+					$strictColor = $cautionColor
+				}
+				Write-Host -nonewline " = = = = = = Submitter "
+				Write-Host -nonewline -f $strictColor "$strictness"
+				Write-Host -nonewline " match "
+				Write-Host -nonewline -f $strictColor "$AuthAccount"
+				Write-Host " = = = = = = "
+			}
+			
+			$wingetOutput = Search-Winget $cleanOut 
+			
+			try {
+				$wgLine = ($wingetOutput | Select-String " $cleanOut ")
+				[System.Version]$wingetVersion = ($wgLine -replace "\s+"," " -split " ")[-2]
+			}catch {
+				$wingetVersion = ""
+			}
+			
+			$titlejoin = ($title -join " ")
+			if (($titlejoin -match "Automatic deletion") -OR ($titlejoin -match "Remove")) {
+				$validColor = "red"
+				$invalidColor = "green"
+				$copyClip = $true
+			}
+			
+			if ($cleanOut -eq "Added") {
+				Write-Host -f $invalidColor "$timevar Error reading package identifier"
+				$noRecord = $true
+			}elseif ($wingetOutput -eq "No package found matching input criteria.") {
+				if ($noNew -eq $true) {
+					$noRecord = $true
+				}
+				Write-Host -f $invalidColor $timevar ($cleanOut) $wingetOutput
+			}elseif ($prVersion -eq "") {
+				$noRecord = $true
+				Write-Host -f $invalidColor "$timevar Error reading PR version"
+			}elseif ($wingetVersion -eq "Unknown") {
+				Write-Host -f $invalidColor "$timevar Error reading Winget version"
+			}elseif ($wingetVersion -eq "input") {
+				$noRecord = $true
+				Write-Host $wingetOutput
+			}elseif ($wingetVersion -eq $null) {
+				Write-Host $wingetOutput
+			}elseif ($wingetVersion -eq "add-watermark") {
+				$noRecord = $true
+				Write-Host -f $invalidColor "$timevar Error reading package identifier"
+			}elseif ($prVersion -gt $wingetVersion) {
```

Comment:
```
Could not compare "20.3" to "". Error: "Cannot convert value "" to type "System.Version". Error: "Version string portion was too short or too long.""
```
Throws error when version isn't parsed correctly - Used #96843 as an example. This should probably be caught and handled cleanly

Reply:
This has been fixed.